assets/src/scripts/monitoring-location/components/daily-value-hydrograph/graph-brush.js

```diff
@@ -73,14 +93,54 @@ export const drawGraphBrush = function(container, store) {
             const group = svg.append('g').attr('class', 'brush')
                 .attr('transform', `translate(${layout.margin.left},${layout.margin.top})`);
 
+            graphBrush.handleSize([1]); // make default handle 1px wide
             graphBrush.extent([[0, 0], [layout.width - layout.margin.right, layout.height - layout.margin.bottom - layout.margin.top]]);
 
+            /* Draws the custom brush handle using an SVG path. The path is drawn twice, once for the handle
+            * on the left hand side, which in d3 brush terms is referred to as 'east' (data type 'e'), and then
+            * inverted for the right hand custom handle. Here 'east' will be a value of either 1 or 0 (in effect, making
+            * it a boolean value of 'east' or 'not east' */
+            const brushResizePath = function(d) {
+                let east = d.type === 'e' ? 1:0,
+                    x = east ? 1 : -1,
+                    y = layoutHeight / 2;
+
+                // Create the svg path using the standard SVG commands M, A, V etc. and substituted variables.
+                return `M ${.5 * x},${y} 
+                    A6,6 0 0 ${east} ${6.5 * x},${y + 6}
+                    V${2 * y - 6}
+                    A6,6 0 0 ${east} ${.5 * x},${2 * y}
+                    Z
+                    M${2.5 * x},${y + 8}
+                    V${2 * y - 8}
+                    M${4.5 * x},${y + 8}
+                    V${2 * y - 8}`;
+            };
+
+            /* Attaches the custom brush handle to the DOM and binds d3 brush data placeholders 'w' for the west end (right side)
+            * and 'e' for east end of the brush area */
+            customHandle = group.selectAll('.handle--custom')
+                .data([{type: 'w'}, {type: 'e'}])
+                .enter().append('path')
+                .attr('class', 'handle--custom')
+                .attr('d', brushResizePath);
+
             // Creates the brush
             group.call(graphBrush);
 
-             // Fill & round corners of brush handles
-            svg.selectAll('.handle').classed('brush-handle-fill', true)
-                .attr('rx',15).attr('ry',15);
+            // Add a class so the default handles can have styling that won't conflict with the slider handle
+            svg.selectAll('.handle').classed('standard-brush-handle', true);
+
+            svg.select('.brush-text-hint').remove();
+            // Add the hint text after the brush is created so that the words sit on top of the brush area
+            svg.call(svg => {
+                svg.append('text')
+                    .classed('brush-text-hint', true)
+                    .text('drag handles to change timeframe')
+                    .call(link(store,(elem, layout) => elem.attr('transform', `translate(${layout.width / 2 + layout.margin.left}, 10)`),
```

Comment:
sorry that I didn't catch this before. Nesting calls to "link" as is done here can cause very strange bugs to appear. I would restructure this to draw the text hint outside the link call on line 84. It only needs to be redrawn when the layout changes so I think it will just be a matter of drawing it after this "link" call. And if that doesn't work to draw it on top, I suggest styling it with a z-index.

Reply:
No problem, I do appreciate your detailed code reviews--hugely helpful. 

Anyway, after going down one dead end after another chasing the exact right combination of position and layer order and finding that SVGs don't like z-index and that the brush area rectangle would redraw over my nicely positioned words when the handles were moved, I finally comprised and moved the text up so that it doesn't overlap at all, which was certainly the easier solution, if not the better looking one as well.
![image](https://user-images.githubusercontent.com/36997804/92653831-8a818300-f2b4-11ea-91fc-3e0fd6cfdcf8.png)